src/blestatemachine.cc

```diff
@@ -559,10 +559,26 @@ namespace BLEPP
 				if(!n.notification())
 					dev.send_handle_value_confirmation();
 			}
+			//client is asking for MTU negotiation, ATTRIBUTE PROTOCOL 3.2.8, 3.4.2 MTU exchange of bluetooth core spec
```

Comment:
At the moment, the current design is to have PDUResponses be for only incoming data. For outgoing data, there are methods on dev which generate the PDUs to send.

Would you be able to change the code so that you had a method so yo could do:

dev.send_<what the name of the method>(current_mtu);

that method would then call internally enc_mtu_resp and could also resize dev's buffer since it has access to that.

Reply:
I put them into bledevice as process_(pdutype)
It's more complex than sending
Unless you want all that logic inside of the state machine case statement - that seems like it would get overwhelming.